### DISCUSSION TITLE
Any attempt to parse shared_album_comments.json in Google Takeout?

### DISCUSSION THREAD

**Discussion by linuxturtle** [simulot/immich-go#433] [Q&A]
Google shared albums have the ability to add comments to a photo, or to the album itself.  This data is shown in a side panel called "Activity", along with thumbnails of the photos.  It also shows up as a little "comment" icon on each photo which has a comment, with a number indicating how many comments the photo has.  This data appears to be exported in a single file at the top of the archive called "shared_album_comments.json".  Has there been any thought or discussion about parsing this file?  I can't find it here or in Issues, if there has been.  I'm fairly new to immich as well, so I'm not really sure if there's even any way to represent it in immich, but it's a cool feature in Google Photos, so if there is, it'd be really cool to preserve the data on import.

**Comment by simulot** [comment]
I don't know about those files. 

At the moment Immich-go can use Album's location, description and texts found in the metadata.json located in the album folder.

**Reply by linuxturtle** [reply]
I've never written anything in go, so the likelihood of me being able to provide a PR is low, but upon some further investigation, it looks like Immich has a 1-1 mapping of this functionality on shared albums.  It would seem that the API is the `Activity` API, where [`createActivity`](https://immich.app/docs/api/create-activity) takes similar arguments as are in the `shared_album_comments.json` file, except I have no clue how to map the `contentUrl` field in the file onto `albumID` in the API.  But it looks like `creationTime` maps directly onto `createdAt`, and `text` maps directly to `comment`.

Does immich-go maintain some kind of mapping between google's `contentUrl` for an album, and the immich API `albumId`?  If so, it would seem pretty easy to parse this file as a post-process sort of thing.

**Reply by linuxturtle** [reply]
Oh, wait, it looks like all the comments are duplicated in the album's metadata.json file under `sharedAlbumComments`, so the `shared_album_comments.json` file is redundant.  I guess immich-go just doesn't know about them?  Would sample data be helpful?

**Reply by simulot** [reply]
Yes indeed. Let's see what we can get

**Reply by linuxturtle** [reply]
Ugh, on further investigation, it looks like only *some* of the comments are in the `metadata.json` file.  Many of the album owner's comments are only in the `shared_album_comments.json` file, so it's not redundant after all.  Oh Google...  Anyway, I opened a feature request #434 to track this issue easier.  Thank you.
